### PR TITLE
Projects!

### PR DESCRIPTION
**Pull request by stazrad**
Resolves: attach_gh_issue_here

### PR Safety Checklist:

 - [x] Added the task to the appropriate release doc under **Enhancements** or **Bug Fixes**
 - [x] Bump `package.json` & `package-lock.json` version numbers to appropriate release
 - [x] (optional) All external API changes have been documented
 - [x] (optional) Build docs: `npm run docs`

### CODE REVIEW

File: src/Project/utils.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      features.forEach(feature => feature.set('_ol_kit_parent', null))
+      const geoJson = new GeoJSON().writeFeatures(features)
+
+      values._ol_kit_project_geojson = geoJson
+    }
+
+    return values
+  })
+  const outputFile = {
+    version,
```
Comment (glenselle) [inline src/Project/utils.js:51]: smart boi

File: src/Project/utils.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  const rawLayers = map.getLayers().getArray()
+  const layers = rawLayers.map(layer => {
+    // some key/values are too large to store in the project file metadata
+    const keysBlacklist = ['source']
+    const keys = layer.getKeys().filter(key => !keysBlacklist.includes(key))
+    const values = {}
+
+    keys.forEach(key => {
+      values[key] = layer.get(key)
+    })
```
Comment (glenselle) [inline src/Project/utils.js:35]: look at using `.getProperties()` instead

Comment (stazrad) [reply src/Project/utils.js:35]: it returns the same internal ol values- sticking with original idea

File: src/Project/utils.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      if (safeValueCheck(value)) values[key] = value
+    })
+    const isVectorLayer = layer instanceof LayerVector || layer.isVectorLayer
+
+    if (isVectorLayer && !layer.get('_ol_kit_data_source')) {
+      // this is a vector layer that was created within ol-kit; get the feature geometries
+      const features = layer.getSource().getFeatures()
+
+      features.forEach(feature => feature.set('_ol_kit_parent', null))
+      const geoJson = new GeoJSON().writeFeatures(features)
```
Comment (stazrad) [inline src/Project/utils.js:39]: options: `{dataProject: map.getView().getProjection(), featureProjection: 'EPSG:4326'}`

File: src/Project/utils.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    const opts = {
+      title: layerData.title
+    }
+
+    if (layerData?._ol_kit_basemap) {
+      // set the basemap
+      loadBasemapLayer(map, layerData._ol_kit_basemap)
+    } else if (layerData?._ol_kit_project_geojson) {
+      // create layer based off geometries
+      const geoJson = new GeoJSON()
```
Comment (stazrad) [inline src/Project/utils.js:78]: options: {featureProjection: map.getView().getProjection(), dataProjection: 'EPSG:4326'}

File: src/Project/utils.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+const readOpts = { dataProjection: 'EPSG:4326', featureProjection: 'EPSG:3857' }
+
+/**
+ * A utility that takes map state and outputs it as a project file
+ * @function
+ * @category Project
+ * @since 1.9.0
+ * @param {Map} - a reference to openlayers map
+ * @returns {File} - a JSON file in ol-kit project format
```
Comment (stazrad) [inline src/Project/utils.js:19]: does not return JSON

File: src/Project/utils.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  return outputFile
+}
+
+/**
+ * A utility that accepts an ol-kit project file and updates the map state
+ * @function
+ * @category Project
+ * @since 1.9.0
+ * @param {Map} - a reference to openlayers map
+ * @param {File} - a JSON file in .olkprj format
```
Comment (stazrad) [inline src/Project/utils.js:75]: this is also not JSON

**Review by glenselle — APPROVED** [review]
LGTM

**Pull request merged**